src/main/java/interface_adaptors/sales_report/SalesReporterController.java

```diff
@@ -0,0 +1,31 @@
+package interface_adaptors.sales_report;
+
+import use_cases.gateway_interfaces.gatewayReaderInterface;
+import use_cases.sales_report.SalesReporterInputBoundary;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SalesReporterController {
+    private final SalesReporterInputBoundary salesReportUseCase;
+    private final gatewayReaderInterface gatewayReader;
+    // I think I need a way to set the filepath here?
+
+
+    public SalesReporterController(SalesReporterInputBoundary useCase, gatewayReaderInterface reader){
+        this.salesReportUseCase = useCase;
+        this.gatewayReader = reader;
+    }
+
+    /**
+     * The controller packages the data and sends it off to the Update Use Case
+     * @param startTime the start of the time range for the sales report
+     * @param endTime the end of the time range for the sales report
+     */
+    public void reportData(String startTime, String endTime) throws IOException {
+        List<String[]> readerData = gatewayReader.getData();
+        salesReportUseCase.generateReport(readerData, startTime, endTime);
```

Comment:
Before passing the data to the use case, maybe package it into a class named "SalesReporterInputData"? The class would just contain three instance variables: readerData, startTime, endTime (the parameters), and help adhere to clean architecture/remove potential code smells.

Reply:
ok, i'll look into this